crowdsec-docs/docs/appsec/troubleshoot.md

```diff
@@ -0,0 +1,266 @@
+DEBU[2023-12-06 15:40:26] Executing disruptive action for rule          action=deny band=inband name=appseclol rule_id=2145145579 type=appsec uuid=adc5ffc4-6080-432c-af93-7c76c79afc25
+DEBU[2023-12-06 15:40:26] Rule matched                                  band=inband name=appseclol rule_id=2145145579 type=appsec uuid=adc5ffc4-6080-432c-af93-7c76c79afc25
+DEBU[2023-12-06 15:40:26] Finish evaluating rule                        band=inband name=appseclol rule_id=2145145579 type=appsec uuid=adc5ffc4-6080-432c-af93-7c76c79afc25
+```
+
+## Ensuring the AppSec Component is running
+
+For the Application Security Component to be running, you need to have:
+ - Acquisition : Acquisition configuration decides on which interface and port is the appsec component running
+ - AppSec Config : appsec config decides which rules are loaded in inband and outofband sections
+ - AppSec Rules : the appsec rules referenced in the appsec config must be installed
+
+Appsec config & rules are obtained by installing the relevant appsec collection, for example `crowdsecurity/appsec-virtual-patching` (`cscli collections install crowdsecurity/appsec-virtual-patching`)
+
+> Example Acquisition config
+
+```yaml
+name: appsec-test
+listen_addr: 127.0.0.1:4242
+appsec_config: crowdsecurity/virtual-patching
+source: appsec
+labels:
+  type: appsec
+```
+
+> Example AppSec config
+
+```yaml
+name: crowdsecurity/virtual-patching
+default_remediation: ban
+#log_level: debug
+inband_rules:
+ - crowdsecurity/vpatch-CVE-2023-40044
+ - crowdsecurity/vpatch-CVE-2023-42793
+# inband_options:
+#  disable_body_inspection: true
+```
+
+> Eample AppSec rule
+
+```yaml
+name: crowdsecurity/vpatch-CVE-2023-42793
+description: "Detect CVE-2023-42793"
+rules:
+  - zones:
+    - URI
+    transform:
+    - lowercase
+    match:
+      type: endsWith
+      value: /rpc2
+labels:
+  type: exploit
+  service: http
+  confidence: 3
+  spoofable: 0
+  behavior: "http:exploit"
+  label: "JetBrains Teamcity auth bypass (CVE-2023-42793)"
+  classification:
+   - cve.CVE-2023-42793
+   - attack.T1595
+   - attack.T1190
+   - cwe.CWE-288
+```
+
+If the appsec component is running, you will see the following in the startup logs:
+
+> rules loading
+
+```
+...
+INFO[2023-12-06 15:06:36] Loaded 0 outofband rules                      component=appsec_config name=crowdsecurity/virtual-patching type=appsec
+INFO[2023-12-06 15:06:36] loading inband rule crowdsecurity/vpatch-CVE-2023-40044  component=appsec_config name=crowdsecurity/virtual-patching type=appsec
+INFO[2023-12-06 15:06:36] loading inband rule crowdsecurity/vpatch-CVE-2023-42793  component=appsec_config name=crowdsecurity/virtual-patching type=appsec
+...
+```
+
+> appsec component runner starting up
+
+```
+INFO[2023-12-06 15:06:36] 1 appsec runner to start                      name=appseclol type=appsec
```

Comment:
Could we just find and replace appseclol with appsec

Reply:
or an different name that clearly showcases name and type are different and eventually that name doesn't have to be appsec.

This is important in my opinion to hint at what people can write or expect. here I guess it's the name of an appsec rule lets call it "testAppsecRule" it's clear and thus easier to read

Reply:
No the name is just the name of acquisition its nothing to do with the rule itself that never gets logged